src/app/guards/guest.guard.ts

```diff
@@ -0,0 +1,18 @@
+import {CanActivateFn, Router} from '@angular/router';
+import {inject} from "@angular/core";
+import {TokenService} from "../services/token.service";
+
+export const guestGuard: CanActivateFn = (route, state) => {
+  const tokenService = inject(TokenService);
+  const router = inject(Router);
+
+  tokenService.isAuthentication.subscribe({
+    next: (value) => {
+      if(value) {
+        router.navigate(['home']);
+      }
+    }
+  })
+
+  return true;
+};
```

Comment:
Was this acknowledged?

Reply:
Bhai, In the code it is still using dependency injection. I guess what samia apu was trying to tell here is to use Constructor injection which is not possible here because `CanActivateFn` is a function.

Reply:
Sure bhai. Understood. Thank you.